src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php

```diff
@@ -99,7 +100,7 @@ public function expand(QuoteTransfer $quoteTransfer): QuoteTransfer
      */
     protected function addUnzerPayment(QuoteTransfer $quoteTransfer): QuoteTransfer
     {
-        $paymentResourceTransfer = $quoteTransfer->getPaymentOrFail()->getUnzerPaymentOrFail()->getPaymentResource();
+        $paymentResourceTransfer = $this->getUnzerPaymentResource($quoteTransfer);
```

Comment:
```suggestion
        $unzerPaymentResourceTransfer = $this->getUnzerPaymentResource($quoteTransfer);
```

Reply:
Resolved in consultation with Ilya Kubanov.